utils/src/crypt.rs

```diff
@@ -320,6 +337,74 @@ impl Cipher {
         out.truncate(count + rest);
         Ok(out)
     }
+
+    pub fn generate_key_for_aes_xts(size: usize) -> Result<Vec<u8>, Error> {
+        let mut buf = vec![0u8; size];
+        if let Err(e) = rand::rand_bytes(&mut buf) {
+            Err(eother!(format!(
+                "failed to generate key for Aes256Xts, {}",
+                e
+            )))
+        } else {
+            Ok(Self::tweak_key_for_xts(&buf).to_vec())
+        }
+    }
+
+    pub fn generate_random_iv(size: usize) -> Result<Vec<u8>, Error> {
+        let mut buf = vec![0u8; size];
+        if let Err(e) = rand::rand_bytes(&mut buf) {
+            Err(eother!(format!("failed to generate iv, {}", e)))
+        } else {
+            Ok(buf)
+        }
+    }
+}
+
+/// Struct to provide context information for data encryption/decryption.
+#[derive(Default, Debug, Clone)]
+pub struct CipherContext {
+    key: Vec<u8>,
+    iv: Vec<u8>,
+    convergent_encryption: bool,
+}
+
+impl CipherContext {
+    /// Create a new instance of [CipherContext].
+    pub fn new(key: Vec<u8>, iv: Vec<u8>, convergent_encryption: bool) -> Result<Self, Error> {
+        if key.len() != 32 {
```

Comment:
The check and the one below, are only for Aes128Xts? If so, please rename this as `new_aes128xts`. `Cipher` is implemented to work for different cipher algorithms. It's confusing `CipherContext` only works for aes128xts. If the limitation is necessary, please make sure to highlight it with proper function names and comments.

Reply:
I'll fix it. THX!